compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12810,6 +12810,235 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
+                          || node->getOpCodeValue() == TR::lucmpeq
+                          || node->getOpCodeValue() == TR::lucmpne
+                          || node->getOpCodeValue() == TR::iucmpeq
+                          || node->getOpCodeValue() == TR::iucmpne;
+
+   bool isSignednessTypeSupported = (!isUnsignedCompare && opNode->chkCannotOverflow())
+           || (isUnsignedCompare && isEqOrNotEqCompare);
+
+   if ((isAddOp || isSubOp)
+         && isSignednessTypeSupported
+         && opNode->getSecondChild()->getOpCode().isLoadConst()
+         && secondChild->getOpCode().isLoadConst()
+         // Only perform the simplification on the first occurance of the add/sub node
+         && (opNode->getFutureUseCount() == opNode->getReferenceCount() - 1))
+      {
+      int64_t oldConst1 = 0, oldConst2 = 0, newConst = 0;     // for signed compare types
+      int64_t signedMax = 0, signedMin = 0;
+
+      uint64_t oldUConst1 = 0, oldUConst2 = 0, newUConst = 0;  // for unsigned compare types
+      uint64_t unsignedMax = 0;
+
+      if (opNode->getOpCode().is1Byte())
+         {
+         oldConst1 = opNode->getSecondChild()->getByte();
+         oldConst2 = secondChild->getByte();
+
+         oldUConst1 = opNode->getSecondChild()->getUnsignedByte();
+         oldUConst2 = secondChild->getUnsignedByte();
+
+         signedMax   = TR::getMaxSigned<TR::Int8>();
+         signedMin   = TR::getMinSigned<TR::Int8>();
+         unsignedMax = TR::getMaxUnsigned<TR::Int8>();
+         }
+      else if (opNode->getOpCode().is2Byte())
+         {
+         oldConst1 = opNode->getSecondChild()->getShortInt();
+         oldConst2 = secondChild->getShortInt();
+
+         oldUConst1 = opNode->getSecondChild()->getUnsignedShortInt();
+         oldUConst2 = secondChild->getUnsignedShortInt();
+
+         signedMax   = TR::getMaxSigned<TR::Int16>();
+         signedMin   = TR::getMinSigned<TR::Int16>();
+         unsignedMax = TR::getMaxUnsigned<TR::Int16>();
+         }
+      else if (opNode->getOpCode().is4Byte())
+         {
+         oldConst1 = opNode->getSecondChild()->getInt();
+         oldConst2 = secondChild->getInt();
+
+         oldUConst1 = opNode->getSecondChild()->getUnsignedInt();
+         oldUConst2 = secondChild->getUnsignedInt();
+
+         signedMax   = TR::getMaxSigned<TR::Int32>();
+         signedMin   = TR::getMinSigned<TR::Int32>();
+         unsignedMax = TR::getMaxUnsigned<TR::Int32>();
+         }
+      else if (opNode->getOpCode().is8Byte())
+         {
+         oldConst1 = opNode->getSecondChild()->getLongInt();
+         oldConst2 = secondChild->getLongInt();
+
+         oldUConst1 = opNode->getSecondChild()->getUnsignedLongInt();
+         oldUConst2 = secondChild->getUnsignedLongInt();
+
+         signedMax   = TR::getMaxSigned<TR::Int64>();
+         signedMin   = TR::getMinSigned<TR::Int64>();
+         unsignedMax = TR::getMaxUnsigned<TR::Int64>();
```

Comment:
It might be good to use a template and `getConst<T>()` to remove this duplication